cli/src/pcluster/schemas/cluster_schema.py

```diff
@@ -237,9 +236,22 @@ class SharedStorageSchema(BaseSchema):
     """Represent the generic SharedStorage schema."""
 
     mount_dir = fields.Str(required=True, validate=get_field_validator("file_path"))
+    name = fields.Str()
+    storage_type = fields.Str(required=True, validate=validate.OneOf(["Ebs", "FsxLustre", "Efs"]))
     ebs = fields.Nested(EbsSchema)
     efs = fields.Nested(EfsSchema)
-    fsx = fields.Nested(FsxSchema, data_key="FsxLustre")
+    fsx = fields.Nested(FsxSchema)
+
+    @pre_load
+    def preprocess(self, data, **kwargs):
+        """Before load the data into schema, change the settings to adapt different storage types."""
+        if data.get("StorageType") == "Efs":
```

Comment:
Let's add a try/except block because data.pop can raise a KeyError if Settings is not passed.
Note that pre_load is executed before validation so we need to be sure that all the requirements are present.

Reply:
Added